packages/app-store/googlecalendar/api/webhook.ts

```diff
@@ -1,48 +1,140 @@
+  delegationCredential: { selectedCalendars: SelectedCalendar[] } | null;
+  userId: number;
+  googleChannelId: string;
+}) {
+  if (selectedCalendarMatchingGoogleChannelId.credential) {
+    // Regular credential case -> There could be multiple selected calendars for the same user because of different eventTypes
+    return selectedCalendarMatchingGoogleChannelId.credential.selectedCalendars;
+  } else if (selectedCalendarMatchingGoogleChannelId.delegationCredential) {
+    // Because same delegationCredential is used by all members of the same organization
+    const selectedCalendarsForAllTheMembersOfTheOrganization =
+      selectedCalendarMatchingGoogleChannelId.delegationCredential.selectedCalendars;
+
+    // There could be multiple selected calendars for the same user because of different eventTypes
+    return selectedCalendarsForAllTheMembersOfTheOrganization.filter(
+      (selectedCalendar) => selectedCalendar.userId === selectedCalendarMatchingGoogleChannelId.userId
+    );
+  } else {
+    throw new HttpError({
+      statusCode: 200,
+      message: `No credential or DelegationCredential found for selected calendar for googleChannelId: ${selectedCalendarMatchingGoogleChannelId.googleChannelId}`,
+    });
   }
+}
 
-  // There could be multiple selected calendars for the same googleChannelId for different eventTypes and same user
+async function updateCacheForGoogleChannel(channelId: string) {
+  // There could be multiple selected calendars for the same googleChannelId for different eventTypes and same user.
   // Every such record has their googleChannel related fields set which are same
   // So, it is enough to get the first selected calendar for this googleChannelId
   // Further code gets all the selected calendars for this calendar's credential
-  const selectedCalendar = await SelectedCalendarRepository.findFirstByGoogleChannelId(
-    req.headers["x-goog-channel-id"]
+  const selectedCalendarMatchingGoogleChannelId = await SelectedCalendarRepository.findFirstByGoogleChannelId(
+    channelId
   );
 
-  if (!selectedCalendar) {
+  if (!selectedCalendarMatchingGoogleChannelId) {
     throw new HttpError({
       statusCode: 200,
-      message: `No selected calendar found for googleChannelId: ${req.headers["x-goog-channel-id"]}`,
+      message: `No selected calendar found for googleChannelId: ${channelId}`,
     });
   }
-  const { credential } = selectedCalendar;
-  if (!credential)
-    throw new HttpError({
-      statusCode: 200,
-      message: `No credential found for selected calendar for googleChannelId: ${req.headers["x-goog-channel-id"]}`,
+
+  const credentialForCalendarService = await getCredentialForCalendarService({
+    ...selectedCalendarMatchingGoogleChannelId,
+    googleChannelId: channelId,
+  });
+
+  const allSelectedCalendarsForCredential = getSelectedCalendarsForGoogleChannel({
+    ...selectedCalendarMatchingGoogleChannelId,
+    googleChannelId: channelId,
+  });
```

Comment:
We need two things to be able to do our job in this file,

credential and allSelectedCalendarsForThatCredential. Abstracted these out to support regular and Delegation Credential